drivers/misc/devmux/devmux.c

```diff
@@ -0,0 +1,180 @@
+		data->selected = index;
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_USERSPACE
+int z_vrfy_devmux_select_set(struct device *dev, size_t index)
+{
+	return z_impl_devmux_select_set(dev, index);
+}
+#include <syscalls/devmux_select_set_mrsh.c>
+#endif
+
+static int devmux_init(struct device *dev)
+{
+	size_t inst = devmux_inst_get(dev);
+	struct devmux_data *const data = dev->data;
+	const struct devmux_config *config = dev->config;
+	size_t sel = data->selected;
+
+	devmux_configs[inst] = config;
+	devmux_datas[inst] = data;
+
+	if (!device_is_ready(config->devs[sel])) {
+		return -ENODEV;
+	}
+
+	*dev = *config->devs[sel];
+
+	return 0;
+}
+
+#define DEVMUX_PHANDLE_TO_DEVICE(node_id, prop, idx)                                               \
+	DEVICE_DT_GET(DT_PHANDLE_BY_IDX(node_id, prop, idx))
+
+#define DEVMUX_PHANDLE_DEVICES(_n)                                                                 \
+	DT_FOREACH_PROP_ELEM_SEP(DT_DRV_INST(_n), devices, DEVMUX_PHANDLE_TO_DEVICE, (,))
+
+#define DEVMUX_SELECTED(_n) DT_INST_PROP(_n, selected)
+
+#define DEVMUX_DEFINE(_n)                                                                          \
+	BUILD_ASSERT(DT_INST_PROP_OR(_n, zephyr_mutable, 0),                                       \
+		     "devmux nodes must contain the 'zephyr,mutable' property");                   \
+	BUILD_ASSERT(DT_INST_PROP_LEN(_n, devices) > 0, "devices array must have non-zero size");  \
+	BUILD_ASSERT(DEVMUX_SELECTED(_n) >= 0, "selected must be > 0");                            \
+	BUILD_ASSERT(DEVMUX_SELECTED(_n) < DT_INST_PROP_LEN(_n, devices),                          \
+		     "selected must be within bounds of devices phandle array");                   \
+	static const struct device *demux_devs_##_n[] = {DEVMUX_PHANDLE_DEVICES(_n)};              \
+	static const struct devmux_config devmux_config_##_n = {                                   \
+		.devs = demux_devs_##_n,                                                           \
+		.n_devs = DT_INST_PROP_LEN(_n, devices),                                           \
+	};                                                                                         \
+	static struct devmux_data devmux_data_##_n = {                                             \
+		.selected = DEVMUX_SELECTED(_n),                                                   \
+	};                                                                                         \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(_n, devmux_init, NULL, &devmux_data_##_n,                            \
+			      (void *)&devmux_config_##_n, PRE_KERNEL_1,                           \
+			      CONFIG_DEVMUX_INIT_PRIORITY, NULL);
+
+DT_INST_FOREACH_STATUS_OKAY(DEVMUX_DEFINE)
+
+#define DEVMUX_DEVICE_GET(_n) DEVICE_DT_GET(DT_DRV_INST(_n)),
+static const struct device *devmux_devices[] = {DT_INST_FOREACH_STATUS_OKAY(DEVMUX_DEVICE_GET)};
```

Comment:
`static const struct device *devmux_devices[] = {DT_INST_FOREACH_STATUS_OKAY(DEVICE_DT_INST_GET)};`

Reply:
Good call

Reply:
Hmm... that one actually results in a compile error, but works with slight modification.